freqtrade/optimize/hyperopt_loss_profit_drawdown.py

```diff
@@ -0,0 +1,29 @@
+"""
+ProfitDrawDownHyperOptLoss
+
+This module defines the alternative HyperOptLoss class based on Profit &
+Drawdown objective which can be used for Hyperoptimization.
+
+Possible to change `DRAWDOWN_MULT` to penalize drawdown objective for
+individual needs.
+"""
+from pandas import DataFrame
+from freqtrade.optimize.hyperopt import IHyperOptLoss
+from freqtrade.data.btanalysis import calculate_max_drawdown
+
+# higher numbers penalize drawdowns more severely
+DRAWDOWN_MULT = 0.075
+
+
+class ProfitDrawDownHyperOptLoss(IHyperOptLoss):
+    @staticmethod
+    def hyperopt_loss_function(results: DataFrame, trade_count: int, *args, **kwargs) -> float:
+        total_profit = results["profit_abs"].sum()
+
+        # from freqtrade.optimize.optimize_reports.generate_strategy_stats()
+        try:
+            _, _, _, _, max_drawdown_per = calculate_max_drawdown(results, value_col="profit_ratio")
```

Comment:
Please don't use `profit_ratio` for drawdown calculations.

While we did use this in the past, we are no longer doing so (except for one legacy usecase), since results are wrong when stake-amount is not identical throughout results (compounding, custom_stake_amount, ...).
Instead, you should use `profit_abs` (you can look at it's usage in `optimize_reports.py`).

Reply:
that makes sense. tried to change with `profit_abs` hope i correctly did